vyper/parser/expr.py

```diff
@@ -586,22 +586,47 @@ def load_bytearray(side):
             raise TypeMismatchException("Unsupported types for comparison: %r %r" % (left_type, right_type), self.expr)
 
     def boolean_operations(self):
-        if len(self.expr.values) != 2:
-            raise StructureException("Expected two arguments for a bool op", self.expr)
-        if self.context.in_assignment and (isinstance(self.expr.values[0], ast.Call) or isinstance(self.expr.values[1], ast.Call)):
-            raise StructureException("Boolean operations with calls may not be performed on assignment", self.expr)
-
-        left = Expr.parse_value_expr(self.expr.values[0], self.context)
-        right = Expr.parse_value_expr(self.expr.values[1], self.context)
-        if not is_base_type(left.typ, 'bool') or not is_base_type(right.typ, 'bool'):
-            raise TypeMismatchException("Boolean operations can only be between booleans!", self.expr)
+        # Iterate through values
+        for value in self.expr.values:
+            # Check for calls at assignment
+            if self.context.in_assignment and isinstance(value, ast.Call):
+                raise StructureException("Boolean operations with calls may not be performed on assignment", self.expr)
+
+            # Check for boolean operations with non-boolean inputs
+            _expr = Expr.parse_value_expr(value, self.context)
+            if not is_base_type(_expr.typ, 'bool'):
+                raise TypeMismatchException("Boolean operations can only be between booleans!", self.expr)
+
+            # TODO: Handle special case of literals and simplify at compile time
```

Comment:
In the interest of keeping changes modular, not sure if this is something that should be implemented as part of this PR, or as a separate PR after this one is merged.

Reply:
I think we should leave this out for now. The plan is to make constant folding it's own step that happens before it even gets to this translation step :)

Reply:
Sounds good to me! 😊 Should Ieave the comment there for now for posterity or want me to pull it out?

Reply:
Leave it, might motivate us to do it sooner.